app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala

```diff
@@ -118,4 +122,32 @@ object BitcoindRpcBackendUtil extends BitcoinSLogger {
       }
     }
   }
+
+  def startBitcoindBlockPolling(
+      wallet: Wallet,
+      bitcoind: BitcoindRpcClient,
+      startCount: Int,
+      interval: FiniteDuration = 10.seconds)(implicit
+      system: ActorSystem,
+      ec: ExecutionContext): Cancellable = {
+    var prevCount = startCount
+    system.scheduler.scheduleWithFixedDelay(0.seconds, interval) { () =>
+      {
+        bitcoind.getBlockCount.flatMap { count =>
+          if (prevCount != count) {
+            val range = prevCount.to(count)
+
+            val hashFs =
+              range.map(bitcoind.getBlockHash(_).map(_.flip))
+            prevCount = count
```

Comment:
I'm not sure if we want to adjust `prevCount` so early. What if `hashFs` fail for whatever reason? 

However this is a competing concern with my suggested change, if `prevCount` is not adjusted until everything is successful, is `10.seconds` enough of an interval to make sure we don't just run this `Runnable`again with the same parameters as last time? I think there should be a note in the code about not making the interval too aggressive.

Also logging in this runnable would be EXTREMELY valuable so that we can see errors happening if there are any. Basically a log everytime it's attempting to run and maybe a log on success/failure?

Reply:
Added logs and docs about the interval.

Reply:
Do you have any thoughts on other things? I.e. moving where we update `prevCount`?

Reply:
I think where `prevCount` is set is correct. I'd rather it be before the futures so we don't need to wait to receive and process all the blocks. Like you said then we might double up and request a set twice.

Reply:
yes but then you never realize you received funds in that case if `hashesF` fails. We probably need utilize some sort of `future.recover()` to reset `prevCount` to the original value in the case that something fails. We should probably also make `prevCount` an `AtomicReference[Int]` to be more async safe.

Reply:
Did both of those